tests/auth_tests/test_tokens.py

```diff
@@ -86,3 +86,32 @@ def test_token_with_different_secret(self):
         # Tokens created with a different secret don't validate.
         self.assertFalse(p0.check_token(user, tk1))
         self.assertFalse(p1.check_token(user, tk0))
+
+    def test_legacy_token_validation(self):
+        """
+        PasswordResetTokenGenerator() is able to validate a token created with
+        a legacy algorithm, typically during a transition period where a new
+        algorithm is introduced.
+        """
+        user = User.objects.create_user('tokentestuser', 'test2@example.com', 'testpw')
+        p_old = PasswordResetTokenGenerator()
+        p_old.algorithm = "sha1"
+        p_old.legacy_algorithm = None
+
+        p_transit = PasswordResetTokenGenerator()
+        p_transit.algorithm = "sha256"
+        p_transit.legacy_algorithm = "sha1"
+
+        p_new = PasswordResetTokenGenerator()
+        p_new.algorithm = "sha256"
+        p_new.legacy_algorithm = None
+
+        tk_old = p_old.make_token(user)
+        self.assertTrue(p_old.check_token(user, tk_old))
```

Comment:
Per https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/ these could use `assertIs(..., True)`. I've done this for existing `.check_token()` tests in #12380.